app/services/trainees/create_from_apply.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module Trainees
+  class CreateFromApply
+    include ServicePattern
+
+    def initialize(json: example_json)
+      @attributes = json["attributes"]
+    end
+
+    def call
+      trainee.save!
+      trainee
+    end
+
+  private
+
+    attr_reader :attributes
+
+    def example_json
+      JSON.parse(example_file)
+    end
+
+    def example_file
+      File.read(Rails.root.join("lib/example_apply_response.json"))
+    end
+
+    def trainee
+      @trainee ||= Trainee.new(mapped_attributes)
+    end
+
+    def mapped_attributes
+      {
+        # TODO: Use actual provider, if found.
+        provider: Provider.find(1),
+        trainee_id: raw_trainee["id"],
+        first_names: raw_trainee["first_name"],
+        last_name: raw_trainee["last_name"],
+        date_of_birth: raw_trainee["date_of_birth"],
+        email: raw_contact_details["email"],
+        # TODO: Use actual route type, if found.
+        record_type: 1,
+        programme_start_date: programme_start_date,
+      }.merge(address)
+    end
+
+    def programme_start_date
+      year, month = raw_course["start_date"].split("-").map(&:to_i)
+      Date.new(year, month)
+    end
+
+    def address
+      raw_contact_details["country"] != "GB" ? international_address : uk_address
+    end
+
+    def international_address
+      {
+        international_address: raw_contact_details["address_line1"],
```

Comment:
I wonder what the none uk contact details shape looks like?

Reply:
This has now changed — it used to be all in one line, now it's split across four. We corrected all the old data to be that shape too